moto/stepfunctions/models.py

```diff
@@ -346,23 +476,22 @@ def update_state_machine(self, arn, definition=None, role_arn=None):
         return sm
 
     def start_execution(self, state_machine_arn, name=None, execution_input=None):
-        state_machine_name = self.describe_state_machine(state_machine_arn).name
-        self._ensure_execution_name_doesnt_exist(name)
-        self._validate_execution_input(execution_input)
-        execution = Execution(
+        state_machine = self.describe_state_machine(state_machine_arn)
+        execution = state_machine.start_execution(
             region_name=self.region_name,
             account_id=self._get_account_id(),
-            state_machine_name=state_machine_name,
             execution_name=name or str(uuid4()),
-            state_machine_arn=state_machine_arn,
             execution_input=execution_input,
         )
-        self.executions.append(execution)
         return execution
 
     def stop_execution(self, execution_arn):
```

Comment:
This should be slightly refactored as well, to `state_machine.stop_execution()`

Reply:
Cool, will do!